### PR TITLE
DEV: update circular average funciton in roi.py

### PR DESCRIPTION
**Pull request by yugangzhang**
@ericdill and @sameera2004  please take a look at the updated circular average funciton in roi.py

### CODE REVIEW

File: skbeam/core/roi.py [hunk 1]
```diff
@@ -403,8 +403,8 @@ def mean_intensity(images, labeled_array, index=None):
     return mean_intensity, index
 
 
-def circular_average(image, calibrated_center, threshold=0, nx=100,
-                     pixel_size=(1, 1),  min_x=None, max_x=None):
```
Comment (tacaswell) [inline skbeam/core/roi.py]: Why break the API?

File: skbeam/core/roi.py [hunk 2]
```diff
@@ -435,12 +428,19 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
...
-    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
-
-    bin_edges, sums, counts = utils.bin_1D(np.ravel(radial_val),
-                                           np.ravel(image), nx,
-                                           min_x=min_x,
-                                           max_x=max_x)
+    radial_val = utils.radial_grid(calibrated_center, image.shape,  pixel_size )
+    
+    if nx is None:
+        ps =np.min( pixel_size )
```
Comment (tacaswell) [inline skbeam/core/roi.py]: follow pep8

File: skbeam/core/roi.py [hunk 3]
```diff
@@ -435,12 +428,19 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
     ring_averages : array
         Radial average of the image. shape is (nx, ).
     """
-    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
-
-    bin_edges, sums, counts = utils.bin_1D(np.ravel(radial_val),
-                                           np.ravel(image), nx,
-                                           min_x=min_x,
-                                           max_x=max_x)
+    radial_val = utils.radial_grid(calibrated_center, image.shape,  pixel_size )
```
Comment (tacaswell) [inline skbeam/core/roi.py]: please remove this change.

File: skbeam/core/roi.py [hunk 4]
```diff
@@ -435,12 +428,19 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
...
+    
+    if nx is None:
+        ps =np.min( pixel_size )
+        max_x = np.max(radial_val)/ps
+        min_x = np.min(radial_val)/ps        
+        nx = int(max_x - min_x)
+    #print (nx)
```
Comment (tacaswell) [inline skbeam/core/roi.py]: remove this line.

File: skbeam/core/roi.py [hunk 5]
```diff
@@ -435,12 +428,19 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
...
+
+    if mask is  None: mask =1
```
Comment (tacaswell) [inline skbeam/core/roi.py]: do not do one-line if statements like this

File: skbeam/core/roi.py [hunk 6]
```diff
@@ -403,8 +403,8 @@ def mean_intensity(images, labeled_array, index=None):
     return mean_intensity, index
 
 
-def circular_average(image, calibrated_center, threshold=0, nx=100,
-                     pixel_size=(1, 1),  min_x=None, max_x=None):
+def circular_average(image, calibrated_center, threshold=-1, nx=None,
+                     pixel_size=None, mask=None):
```
Comment (tacaswell) [inline skbeam/core/roi.py]: Please document the new kwarg

**Comment by tacaswell** [comment]
Please add a non-trivial test for the new kwarg.

**Comment by yugangzhang** [comment]
I already did  test on chx.

**Comment by tacaswell** [comment]
The tests need to be in the code base so they get run as part of the test suite.

**Comment by sameera2004** [comment]
@yugangzhang As I remember, you want to have options to give min_x and max_x for the bin_centers. That way you can give the first q_ring and last q_ring values and get the circular average only on that region. I already added that. Did you try to use the updated circular_average?  Please see the attachment
screen shot 2016-02-09 at 9 30 58 pm

**Comment by ericdill** [comment]
@yugangzhang @sameera2004 If I understand correctly, the proposed functionality change in this PR already exists in scikit-beam. Is that correct?

**Comment by yugangzhang** [comment]
Maybe not include my latest version. @ sameera can you double check?

**Comment by ericdill** [comment]
@yugangzhang; to use the '@' mentions, you need to put the '@' right before the username with no space. '@ sameera' will not send her an email update. However, '@sameera2004' will.

**Comment by yugangzhang** [comment]
Thanks! I knew but I just a smart iPhone, it is not the supposed smart.

**Comment by ericdill** [comment]
Ah, I understand. Sorry your iphone is not that smart :sad:

**Comment by sameera2004** [comment]
@ericdill @yugangzhang yes it is included. @yugangzhang on Monday I will show you how to use it.

**Comment by ericdill** [comment]
Ok great. Thanks @sameera2004 . @yugangzhang please reopen if there is functionality that we do not have in scikit-beam already. thanks

**Pull request closed without merging**